contracts/DLCManager.sol

```diff
@@ -352,14 +395,19 @@ contract DLCManager is
             revert DepositTooSmall(amountToLockDiff, minimumDeposit);
         }
 
+        // We fetch the current total minted value in all vaults before we update this Vault
+        uint256 currentTotalMinted = getTotalValueMintedInVaults();
+
         dlc.fundingTxId = btcTxId;
         dlc.wdTxId = "";
         dlc.status = DLCLink.DLCStatus.FUNDED;
 
         dlc.valueLocked = newValueLocked;
         dlc.valueMinted = newValueLocked;
 
-        _mintTokens(dlc.creator, amountToMint);
+        if (_checkMint(amountToMint, currentTotalMinted)) {
```

Comment:
if this check is specifically just for CL_POR i would say you can name it that way? If you want to put other things into check_mint, then it can be called check_mint but then the PoR code can be in another sub function called PoR check